modules/040-node-manager/images/caps-controller-manager/src/api/infrastructure/v1alpha1/staticmachinetemplate_webhook.go

```diff
@@ -62,8 +62,19 @@ func (r *StaticMachineTemplate) ValidateCreate() (admission.Warnings, error) {
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *StaticMachineTemplate) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
 	oldStaticMachineTemplate := old.(*StaticMachineTemplate)
-	if !reflect.DeepEqual(r.Spec, oldStaticMachineTemplate.Spec) {
-		err := field.Forbidden(field.NewPath("spec"), "StaticMachineTemplate.spec is immutable")
+
+	// Allow changes to labelSelector as it's a valid operation for NodeGroup management
+	// Create copies to compare without labelSelector
+	newSpecCopy := r.Spec.DeepCopy()
```

Comment:
hm, here we https://github.com/deckhouse/deckhouse/pull/17276/changes#diff-04b6f25bfbdd136bc5bee7dadedb4b93e54901d3f6029a892e82550a197647daR1680 add warning that says client does not change label selector, but here we fill nil label selectors and we do not validate this case because label selectors always be equal (both is nil)

In this situation we can use this algorithm:

if oldSpecCopy.Template.Spec.LabelSelector is nil or empty (oldSpecCopy.Template.Spec.LabelSelector.MatchLabels is empty and oldSpecCopy.Template.Spec.LabelSelector.MatchExpressions is empty) - allow change, otherwise use `reflect.DeepEqual(newSpecCopy, oldSpecCopy)`